pkg/kapp/clusterapply/waiting_changes.go

```diff
@@ -126,3 +139,12 @@ func (c *WaitingChanges) Complete() error {
 func (c *WaitingChanges) stats() string {
 	return fmt.Sprintf("[%d/%d done]", c.numWaited, c.numTotal)
 }
+
+func checkForTimeout(change WaitingChange, timeout time.Duration, err error) error {
+	if time.Now().Sub(change.startTime) > timeout {
+		return errors.New("resource wait timeout")
```

Comment:
this is where should build full error message instead of doing it later (e.g. fmt.Errorf("Resource timed out waiting after %s", c.opts.ResourceWaitTimeout))

Reply:
Done